workers/generic-worker/errorreport/errorreport_test.go

```diff
@@ -50,12 +51,12 @@ func TestSendEmptyExtra(t *testing.T) {
 
 	lock.Lock()
 	assert.True(t, errorReported, "No error-report was received")
-	lock.Unlock()
 }
 
 func TestSendExtra(t *testing.T) {
 	errorReported := false
 	lock := sync.Mutex{}
+	defer lock.Unlock()
```

Comment:
Same here, this is the wrong place to add this defer statement, since any failure after this line, but before line 83 would lead to a runtime exception.